docs/release.md

```diff
@@ -97,7 +98,9 @@ merged to `main` since the currently-deployed release was tagged.
 
 In this situation, we create a new hotfix release branch starting at the point
 of the previous release tag. We then use the normal GitHub PR and code-review
-process to merge the necessary fix(es) to the branch. Finally we create a new release tag at the tip of the release branch instead of the tip of main.
+process to merge the necessary fix(es) to `main` first. The fix commits are then
+cherry-picked onto the release branch, typically via another pull request for
+visibility. Finally we create a new release tag at the tip of the release branch instead of the tip of main.
```

Comment:
There's been a slight misunderstanding here: in the existing text, "the branch" means the _release_ branch, not the main branch. Specifically, we are no longer willing (or able, thanks to GitHub branch protection rules) to cherry-pick fixes directly to the release branch. They _must_ be landed on that branch via the normal pull-request process, just with the "target branch" set to to release branch instead of main.